Objects/longobject.c

```diff
@@ -5542,11 +5542,14 @@ Return an array of bytes representing an integer.
 static PyObject *
 int_to_bytes_impl(PyObject *self, Py_ssize_t length, PyObject *byteorder,
                   int is_signed)
-/*[clinic end generated code: output=89c801df114050a3 input=ddac63f4c7bf414c]*/
+/*[clinic end generated code: output=89c801df114050a3 input=d6b9571b1ee1fd05]*/
 {
     int little_endian;
     PyObject *bytes;
 
+    if (byteorder == NULL)
+        byteorder = PySys_GetObject("byteorder");
+
     if (_PyUnicode_EqualToASCIIId(byteorder, &PyId_little))
```

Comment:
This should be a tad faster:

```suggestion
        little_endian = PY_LITTLE_ENDIAN;
    else if (_PyUnicode_EqualToASCIIId(byteorder, &PyId_little))
```

Reply:
My first reaction was "what if `sys.byteorder` was `'big'`?", but given this in `sysmodule.c`:

```
#if PY_BIG_ENDIAN
    SET_SYS_FROM_STRING("byteorder", "big");
#else
    SET_SYS_FROM_STRING("byteorder", "little");
#endif
```

I feel like the performance improvement will be negligible and I'd rather opt for readability over a few cycles (yeah, I could add a comment, but still).

Reply:
Hm... then at the very least, the return value of `PySys_GetObject("byteorder")` should be checked for `NULL`:

```
>>> import sys
>>> del sys.byteorder
>>> (42).to_bytes()
python: Objects/unicodeobject.c:497: _PyUnicode_CheckConsistency: Assertion `op != NULL' failed.
Aborted (core dumped)
```

Reply:
> I feel like the performance improvement will be negligible and I'd rather opt for readability over a few cycles (yeah, I could add a comment, but still).

I wouldn't be surprised if this lookup accounted for a significant portion of this function's runtime. I'll run a few quick benchmarks to confirm my intuition, though.

Reply:
On an isolated CPU with PGO/LTO... using `sys.byteorder`:

```
$ sudo $(which pyperf) timeit --rigorous 'list(map(int.to_bytes, range(256)))'
.........................................
Mean +- std dev: 31.5 us +- 1.3 us
```

Using `Py_LITTLE_ENDIAN`:

```
$ sudo $(which pyperf) timeit --rigorous 'list(map(int.to_bytes, range(256)))'
.........................................
Mean +- std dev: 9.32 us +- 0.27 us
```

It appears that well over 70% of the time in the single-byte cases is spent looking up `sys.byteorder`.

Reply:
Thanks for doing that analysis @brandtbucher - it's compelling.  If you want to resuggest the change but with a comment explaining why it works (so future readers don't have to scratch their head), I'll accept it.

Reply:
Here you go:

```suggestion
        // PY_LITTLE_ENDIAN is 1 on little-endian systems, and 0 on big-endian
        // systems. In the default case, it's significantly faster than looking
        // up sys.byteorder and performing the string comparisons below:
        little_endian = PY_LITTLE_ENDIAN;
    else if (_PyUnicode_EqualToASCIIId(byteorder, &PyId_little))
```